test/functional/fixtures/before-test.js

```diff
@@ -0,0 +1,146 @@
+var createTestCafe         = require('../../../lib');
+var browserNatives         = require('testCafe-browser-natives');
+var SauceLabsConnector     = require('saucelabs-connector');
+var Promise                = require('pinkie');
+var caller                 = require('caller');
+var path                   = require('path');
+var os                     = require('os');
+var Site                   = require('../site');
+var getErrorByBrowsersInfo = require('../helpers/report-helper.js');
+var config                 = require('../config.js');
+
+var testCafe     = null;
+var browsersInfo = null;
+
+var slConnector = null;
+var slBrowsers  = null;
+
+var SAUCELABS_SETTINGS = {
+    username:  process.env.SAUCE_USERNAME,
+    accessKey: process.env.SAUCE_ACCESS_KEY,
+    build:     process.env.TRAVIS_JOB_ID || '',
+    tags:      [process.env.TRAVIS_BRANCH || 'master'],
+    name:      'testcafe functional tests'
+};
+
+before(function () {
+    var hostname = process.env.TRAVIS ? os.hostname() : config.testCafe.hostname;
+
+    return createTestCafe(hostname, config.testCafe.port1, config.testCafe.port2)
+        .then(function (tc) {
+            var startPromise        = null;
+            var openBrowserPromises = null;
+
+            testCafe = tc;
+
+            browsersInfo = config.browsers
+                .filter(function (browser) {
+                    return SAUCELABS_SETTINGS || !browser.remoteOnly;

```

Comment:
I've tried this way, which you recommend, before, but eslint showed error Unexpected 'else' after 'return'.